Shift3/boilerplate-client-angular
# feat,refactor(user profile): redesigns user profile page & implements user email change workflow

## Changes
  1. Redesigns user profile page to render two separate forms.
  2. Notifies users if an email change request already exists & allows them to resend verification email.
  3. Creates a new auth route to render a form where the user can input the verification code received in the email.
  4. Refactors `SaveCancelConfig` to allow hiding the cancel button in some forms.
  5. Refactors `FormInput` component to validate input value & set to int value if input type is number.
  6. Hides `Edit` button on the self user in the user table.
  7. Hides `email` field from the self user update form.
  8. Fixes the user profile dropdown menu position on both navigation bar positions.

## Purpose
Redesign the user profile page to have separate forms to manage user profile & email. Also to create a workflow for user email change.

## Testing
- Pull in the changes to your local copy of this branch and restart Docker.
- Log into the application.
- Go to the `User Profile` page.
- Verify that 2 forms are displayed as below:
  ![Screen Shot 2022-02-04 at 3 44 58 PM](https://user-images.githubusercontent.com/20917683/152617705-8be4592b-bcf8-46ef-a5e3-c1a83359b6bc.png)
- Request an email change by entering it in the `Update Email` form. 
- Verify that a toast message is displayed as shown below. Also, verify that the `Update Email` form section contains some additional text as shown below:
  ![Screen Shot 2022-02-04 at 3 46 06 PM](https://user-images.githubusercontent.com/20917683/152617792-d5138e09-c76e-4b0f-b4a9-163dfc5a4e39.png)
- Verify that the `Update Email` form section displays text notifying that a change request is already in place like shown below:
  ![Screen Shot 2022-02-04 at 3 49 12 PM](https://user-images.githubusercontent.com/20917683/152618119-c7cbafbd-63de-40e2-915c-39d0b6b69543.png)
- Verify that you are able to continue to use the application without getting logged out.
- Look for the email change verification email in the server's terminal or at `http://localhost:3000/emails`.
- Click the Confirm Email Change link from the email.
- Verify that a confirmation form with input to enter the verification code is displayed as shown below:
  ![Screen Shot 2022-02-04 at 3 48 00 PM](https://user-images.githubusercontent.com/20917683/152617941-1933680d-2cbc-4c5b-9e92-3094dbe105e6.png)  
- Verify that the value of the input is 0 by default with an error message & the button is disabled.
- Enter any random 6 digit number in the field & submit. Verify that an error message is thrown.
- Copy the verification code from the email and enter it into the form. Verify that a success message is displayed & you are navigated to the login page.
- On the app attempt to log in with the old email & verify that an error is thrown.
- Login with the new email & verify that the additional text in the Update Email form is no longer displayed.

Closes #367
